cleanlab/internal/multilabel_utils.py

```diff
@@ -0,0 +1,287 @@
+        or (y.dtype.kind == "f" and bool(np.all(y.astype(int) == y)))
+    )
+
+
+class _Wrapper:
+    """Helper class for wrapping callable functions as attributes of an Enum instead of
+    setting them as methods of the Enum class.
+
+
+    This class is only intended to be used internally for the ClassLabelScorer or
+    other cases where functions are used for enumeration values.
+    """
+
+    def __init__(self, f: Callable) -> None:
+        self.f = f
+
+    def __call__(self, *args, **kwargs):
+        return self.f(*args, **kwargs)
+
+    def __repr__(self):
+        return self.f.__name__
+
+
+class ClassLabelScorer(Enum):
+    """Enum for the different methods to compute label quality scores."""
+
+    SELF_CONFIDENCE = _Wrapper(get_self_confidence_for_each_label)
+    NORMALIZED_MARGIN = _Wrapper(get_normalized_margin_for_each_label)
+    CONFIDENCE_WEIGHTED_ENTROPY = _Wrapper(get_confidence_weighted_entropy_for_each_label)
+
+    def __call__(self, labels: np.ndarray, pred_probs: np.ndarray, **kwargs) -> np.ndarray:
+        """Returns the label-quality scores for each datapoint based on the given labels and predicted probabilities."""
+        return self.value(labels, pred_probs, **kwargs)
+
+
+class MultilabelScorer:
+    """Aggregates label quality scores across different classes to produce one score per example in multi-label classification tasks."""
+
+    def __init__(
+        self,
+        base_scorer: ClassLabelScorer = ClassLabelScorer.SELF_CONFIDENCE,
+        aggregator: Optional[Callable[..., np.ndarray]] = None,
+        *,
+        strict: bool = True,
+    ):
+        """
+        Initialize object with a base scoring function that is applied to each label and function that pools scores accross labels.
+
+        Parameters
+        ----------
+        base_scorer:
+            A function that computes a quality score for a single label in a multi-label classification problem.
+
+        aggregator:
+            A function that aggregates the scores computed by base_scorer over all labels.
+            If None, the scores are averaged.
+
+        strict:
+            If True, raises an error if the labels are not binary or are incompatible with the predicted probabilities.
+
+        Examples
+        --------
+        >>> from cleanlab.internal.multilabel_utils import MultilabelScorer, ClassLabelScorer
+        >>> import numpy as np
+        >>> scorer = MultilabelScorer(
+        ...     base_scorer = ClassLabelScorer.NORMALIZED_MARGIN,
+        ...     aggregator = np.min,
+        ... )
+        >>> labels = np.array([[0, 1, 0], [1, 0, 1]])
+        >>> pred_probs = np.array([[0.1, 0.9, 0.1], [0.4, 0.1, 0.9]])
+        >>> scores = scorer(labels, pred_probs)
+        >>> scores
+        array([0.9, 0.4])
+        """
+        self.base_scorer = base_scorer
+        if aggregator is None:
+            self.aggregator = np.mean
+        else:
+            self.aggregator = aggregator
```

Comment:
I think type check is failing somewhere around here FYI

Reply:
Weirdly, the trick is to annotate the first assignment in cases like this.
The inferred type of `np.mean` doesn't quite match `Callable[..., np.ndarray]`, but the annotation fixes the failed check.